### PR TITLE
Modified scraper

### PR DESCRIPTION
**Pull request by jian-li1**

### CODE REVIEW

File: backend/src/Web/WebCrawler.py [hunk 1]
```diff
@@ -28,15 +31,24 @@ class LinkResolver:
     """Handles URL resolution and filtering of links on a page."""
 
     @staticmethod
-    def resolve_links(base_url, soup, visited):
+    def resolve_links(url, base_url, soup, visited):
```
Comment (morshoto) [inline backend/src/Web/WebCrawler.py:47]: Can we make overload function instead of adjusting existing function?

**Comment by jian-li1** [comment]
Completed, ready to merge

**Comment by shoot649854** [comment]
@jian-li1 
- Instead of modifying the existing `resolve_links` function, could we create an overload function to handle the additional parameter? This might help maintain backward compatibility.
- Also, there are some conflicts, particularly in `backend/app.py`. Please resolve these to ensure a smooth merge process.

Thanks!

**Review by morshoto — CHANGES_REQUESTED** [review]
@jian-li1 
- Instead of modifying the existing `resolve_links` function, could we create an overload function to handle the additional parameter? This might help maintain backward compatibility.
- Also, there are some conflicts, particularly in `backend/app.py`. Please resolve these to ensure a smooth merge process.

Thanks!

**Comment by jian-li1** [comment]
I've addressed the changes that you proposed, let me know if you have any other comments/concerns.

**Review by morshoto — APPROVED** [review]
LGTM, ready to merge 🔥

**Pull request merged**